pkg/controllers/clusterresourceplacement/resource_selector_test.go

```diff
@@ -807,6 +814,886 @@ func createResourceContentForTest(t *testing.T, obj interface{}) *fleetv1beta1.R
+				}
+			}(),
+			wantError: controller.ErrUnexpectedBehavior,
+		},
+		{
+			name:          "should return error using label selector when informer manager returns error (getting namespace) for cluster scoped placement",
+			placementName: types.NamespacedName{Name: "test-placement"},
+			selectors: []fleetv1beta1.ClusterResourceSelector{
+				{
+					Group:          "",
+					Version:        "v1",
+					Kind:           "Namespace",
+					SelectionScope: fleetv1beta1.NamespaceWithResources,
+				},
+			},
+			resourceConfig: utils.NewResourceConfig(false), // allow all resources
+			informerManager: func() *testinformer.FakeManager {
+				return &testinformer.FakeManager{
+					IsClusterScopedResource: true,
+					Listers: map[schema.GroupVersionResource]*testinformer.FakeLister{
+						namespaceGVR: {
+							Objects: []runtime.Object{},
+							Err:     apierrors.NewNotFound(schema.GroupResource{Group: "", Resource: "namespaces"}, "test-ns"),
+						},
+					},
+				}
+			}(),
+			wantError: controller.ErrAPIServerError,
+		},
+		{
+			name:          "should return error when informer manager returns non-NotFound error (getting deployment) for cluster scoped placement",
+			placementName: types.NamespacedName{Name: "test-placement"},
+			selectors: []fleetv1beta1.ClusterResourceSelector{
+				{
+					Group:          "",
+					Version:        "v1",
+					Kind:           "Namespace",
+					Name:           "test-ns",
+					SelectionScope: fleetv1beta1.NamespaceWithResources,
+				},
+			},
+			resourceConfig: utils.NewResourceConfig(false), // allow all resources
+			informerManager: func() *testinformer.FakeManager {
+				return &testinformer.FakeManager{
+					IsClusterScopedResource: true,
+					Listers: map[schema.GroupVersionResource]*testinformer.FakeLister{
+						namespaceGVR: {Objects: []runtime.Object{testNamespace, prodNamespace, testDeletingNamespace, kubeSystemNamespace}},
+						deploymentGVR: {
+							Objects: []runtime.Object{},
+							Err:     errors.New("connection timeout"),
+						},
+					},
+					NamespaceScopedResources: []schema.GroupVersionResource{deploymentGVR},
+				}
+			}(),
+			wantError: controller.ErrUnexpectedBehavior,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			r := &Reconciler{
+				ResourceConfig:  tt.resourceConfig,
+				InformerManager: tt.informerManager,
+				RestMapper:      newFakeRESTMapper(),
+			}
+
+			got, err := r.gatherSelectedResource(tt.placementName, tt.selectors)
+			if gotErr, wantErr := err != nil, tt.wantError != nil; gotErr != wantErr || !errors.Is(err, tt.wantError) {
+				t.Fatalf("gatherSelectedResource() = %v, want error %v", err, tt.wantError)
+			}
+			if tt.wantError != nil {
+				return
+			}
+			if diff := cmp.Diff(tt.want, got); diff != "" {
+				t.Errorf("gatherSelectedResource() mismatch (-want +got):\n%s", diff)
+			}
+		})
+	}
+}
+
+// fakeRESTMapper is a minimal RESTMapper implementation for testing
+type fakeRESTMapper struct {
```

Comment:
Move these to test/utils too?

Reply:
we can do if we find other controllers need this.